apps/portal/app/components/atom-search-combobox.tsx

```diff
@@ -0,0 +1,189 @@
+  id,
+  name,
+  avatarSrc,
+  value,
+  onClick,
+  onSelect,
+  attestors,
+}: AtomSearchComboboxItemProps) => {
+  return (
+    <CommandItem
+      key={id}
+      className="border border-transparent aria-selected:bg-primary/5 aria-selected:text-primary hover:border-y-border/10 px-2.5 py-1.5 cursor-pointer [&:not(:last-child)]:border-b-border/10 group"
+      onClick={onClick}
+      onSelect={onSelect}
+    >
+      <div className="flex items-center justify-between w-full gap-8">
+        <div className="flex items-center gap-2 min-w-0">
+          {avatarSrc ? (
+            <img
+              src={avatarSrc}
+              alt={name}
+              className="h-8 w-8 flex-shrink-0 rounded-full object-cover"
+            />
+          ) : (
+            <div className="h-8 w-8 flex-shrink-0 rounded-full bg-foreground/10" />
+          )}
+          <div className="flex items-center gap-1 min-w-0 flex-1">
+            <div className="transition-colors text-md font-medium group-hover:text-foreground truncate">
+              {name}
+            </div>
+            {id !== undefined && (
+              <div className="flex-shrink-0 flex items-center gap-1 bg-foreground/10 rounded-md py-0.5 px-1.5">
+                <div className="transition-colors text-sm text-foreground/70 group-hover:text-foreground font-medium">
+                  {id}
+                </div>
+              </div>
+            )}
+          </div>
+        </div>
+        <div className="transition-colors flex items-center gap-1 bg-foreground/10 rounded-md py-0.5 px-2 text-foreground/70 group-hover:text-foreground flex-shrink-0">
+          <div className="text-sm font-medium flex items-center gap-0.5">
+            <Icon name={IconName.eth} className="w-3 h-3 text-foreground" />
+            {value?.toFixed(3)}
+          </div>
+          <span className="h-[2px] w-[2px] bg-foreground/50" />
+          <div className="flex items-center gap-0.5">
+            <Icon name={IconName.people} className="w-3 h-3 text-foreground" />
+            {attestors}
+          </div>
+        </div>
+      </div>
+    </CommandItem>
+  )
+}
+
+export interface AtomSearchComboboxProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  onAtomSelect?: (atom: GetAtomQuery['atom']) => void
+  onCreateAtomClick?: () => void
+  initialValue?: string
+  placeholder?: string
+}
+
+export function AtomSearchCombobox({
+  onAtomSelect = () => {},
+  initialValue = '',
+  placeholder = 'Search for an atom...',
+  ...props
+}: AtomSearchComboboxProps) {
+  const [searchValue, setSearchValue] = React.useState(initialValue)
+  const debouncedSearch = useDebounce(searchValue, 300)
+  const [isOpen, setIsOpen] = React.useState(false)
+  const [searchResults, setSearchResults] = React.useState<
+    Omit<
+      NonNullable<GetAtomQuery['atom']>,
+      'asSubject' | 'asPredicate' | 'asObject'
+    >[]
+  >([])
+
+  const { data: atomsData, isLoading } = useGetAtomsQuery(
+    {
+      where: {
```

Comment:
Great! I'm glad this is working -- thanks for pulling the debounce into it's own hook. This is how I approached it in the playground tests so I'm glad it works and that you abstracted it into a hook!